ts/RootContainer.tsx

```diff
@@ -128,7 +134,16 @@ class RootContainer extends React.PureComponent<Props> {
         )}
         {shouldDisplayVersionInfoOverlay && <VersionInfoOverlay />}
         <Navigation />
-        <IdentificationModal />
+        {isVersionAppSupported(
```

Comment:
What do you think to refactor the render to make it simple and more reabable?

```typescript
  public render() {
    // FIXME: perhaps instead of navigating to a "background"
    //        screen, we can make this screen blue based on
    //        the redux state (i.e. background)
    const isAppOutOfDate = !isVersionAppSupported(
      this.props.backendInfo !== undefined
        ? this.props.backendInfo.minAppVersion
        : undefined,
      DeviceInfo.getVersion()
    );
    return (
      <Root>
        <StatusBar barStyle="dark-content" />
        {Platform.OS === "android" && (
          <FlagSecureComponent
            isFlagSecureEnabled={!this.props.isDebugModeEnabled}
          />
        )}
        {shouldDisplayVersionInfoOverlay && <VersionInfoOverlay />}
        <IdentificationModal />
        <Navigation />
        <IdentificationModal />
        {isAppOutOfDate && <UpdateAppModal />}
        <LightModalRoot />
      </Root>
    );
  }
}
```

Reply:
@Undermaken Ok but I left control over the modals `{isAppOutOfDate ? <UpdateAppModal /> : <IdentificationModal />}` so as to avoid a possible overlap of the two